### ISSUE TITLE
Incomplete sync

### ISSUE THREAD

**Issue by surfatwork** [nylas/N1#2317]
I have set up a local N1 sync-engine on a local server, which pulls data off a local dovecot imap server on the same server. The local imap serves up a large local maildir store (~13GB). The server exposes the nylas api (x.x.x.x:5555) through a public url, and also locally. 
The N1 client on my laptop accesses the sync-engine through the internet i.e. through a public URL and not over my home 192.168.x.x network.  This is to ensure that the setup works when I am not at home.

I have confirmed that the imap folders are served properly and fully by connecting via another mail client (evolution). All messages in all mail folders are displayed properly through the public URL and 
on the home network.

**Issue:**
The N1 client does not display all my mails. Mails in some folders are synced and displayed. Other folders seem to be "not synced"/ignored. The folder name is displayed in the left pane, but clicking on the folder only produces a "Nothing to Display" in the Message List window.
I also kept the inbox-api console window open on the server. Clicking on an "ignored" folder produces no console output from inbox-api. Clicking on a "not-ignored" folder produces console chatter, say to download an attachment, and also results in the N1 client displaying mails correctly.
I have tested this via my local network too, to rule out any throttling on external networks. Same result.

##### Are there any related issues?
Not that I could see

##### What operating system are you using?
4.5.4-1-ARCH on both server and desktop

##### What version of N1 are you using?
0.4.40-4007743

**Bug?**
##### Do you have any third-party plugins installed?
Yes. Quick replies

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
Yes. see above

##### Is the issue reproducible with a particular attachment, message, signature, etc?

This is an example that N1 doesnt display messages in some folders. In the screenshot below, you can see that Evolution displays mails from the Sent_10_11 folder, while N1 claims that there is nothing to display. Both are connected to the same user on the same imap server.
![screenshot from 2016-05-25 10-23-25](https://cloud.githubusercontent.com/assets/7880295/15534899/016865c8-2263-11e6-9741-460c9930966f.png)

This is an example that N1 does display messages in some folders. In the screenshot below, you can see that both Evolution and N1 are correctly displaying messages from the selected folder.
![screenshot from 2016-05-25 10-32-30](https://cloud.githubusercontent.com/assets/7880295/15535182/276e7b76-2264-11e6-9250-dbcbab57292c.png)
...

**Comment by bengotow** [comment]
Hey! This is super interesting...  can you try deleting N1's local cache of your mailbox at `~/.nylas/edgehill.db`? When you re-launch the app it will pull down things from your local copy of the sync engine again. If you're still missing mail after that, we'll definitely want to investigate this further!

**Comment by surfatwork** [comment]
Hi, I tried re-syncing. renamed .nylas folder to .nylas_old, created a new .nylas folder, copied over the config.json and fired up N1.
Unfortunately, same result - stops after a while. seems to be at the same place - the same folders are ignored. Maybe it's barfing on some particular message? will check this further.

Please let me know what info you need from me. Meanwhile, I'll keep investigating.

If it matters at all (dont think it should), the .nylas folder is on another ext4 partition (with plenty of space), and it's bind mounted to ~/.nylas . I dont have enough space on /home, and also this allows me to multi-boot linux installs.

**Comment by bengotow** [comment]
Thanks @surfatwork that's really interesting. Can you look in Window > Activity and see if any failures are shown in red there as it syncs for the first time? It sounds like this may actually be an issue on the sync-engine side, but I'm not sure why it wouldn't pick up certain folder contents.

**Comment by surfatwork** [comment]
Hi,
Sorry for being AWOL. got caught up in work.
So I recreated an empty ~/.nylas folder, copied over config.json and fired up the client. The result was the same i.e an incomplete sync.

1. I kept the inbox-api running in a console. and this popped up: 
`[WARNING] {"endpoint": "namespace_api.thread_query_api", "account_id": 3, "level": "warning", "timestamp": "2016-05-30T19:34:29.183037Z", "module": "inbox.models.thread:83", "thread_id": 13244, "greenlet_id": 111544272, "event": "Thread does not have associated messages"}`

There were also warnings from the contact api, but I am assuming those aren't relevant since I dont have any contacts to be synced from this server.

2. I also kept the Developer window in the client open. and this popped up:
![screenshot from 2016-05-30 20-41-13](https://cloud.githubusercontent.com/assets/7880295/15657224/e0dc8260-26a6-11e6-9d79-60199a157731.png)

Can I share any other info?
